### ISSUE TITLE
Snap not running on Ubuntu 20.04

### ISSUE THREAD

**Issue by siauderman** [electron-userland/electron-builder#5067]
* **Version**: 21.2.0

* **Target**: Snap

Hi, my application works fine on all other platforms. However, when I installed it via Snap, running the application by using the snap installed shortcut in /snap/bin does not work and neither does running it from the KDE menu item installed by Snap (I presume it would be the same in GNOME as well although I haven't tested it). However, going into the installed folder (/snap/covidgame/current) and running /snap/covidgame/current/covidgame works.

I'm not sure what is wrong but I don't think it's my app or electron since it is otherwise working on all other platforms (and in fact the installation works, just the symlink and the menu shortcut are broken). I think there might be something linked wrongly in the built Snap file.

My game is located at https://snapcraft.io/covidgame

**Comment by srevinsaju** [comment]
Reproducible, Happens just the same for me

**Comment by KhushrajRathod** [comment]
I have the same problem, snaps don't work without --devmode

**Comment by KhushrajRathod** [comment]
This could be due to the fact that snaps cannot access /tmp/

**Comment by siauderman** [comment]
> Is this still relevant? If so, what is blocking it? Is there anything you can do to help move it forward?
> 
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Yes, it is still relevant. My snaps built using electron-builder are still not working.

**Comment by siauderman** [comment]
This is not a similar issue, my snaps can install but they cannot run.

**Comment by siauderman** [comment]
Hi, any possibility that this can be fixed or anyone found a workaround?

**Comment by siauderman** [comment]
Hi can somebody help with this please? Any workaround?

**Comment by amialive** [comment]
We are facing the same issue. The app only runs when executing the following command: `/snap/<app_name>/current/<app_name>`

**Comment by siauderman** [comment]
Yes, still relevant. Why is this still not fixed yet?

**Comment by siauderman** [comment]
@srevinsaju @khrj @amialive Any fix? I've just tried with the latest electron-builder and it's still not working unless I run /snap/<app_name>/current/<app_name>.

**Comment by srevinsaju** [comment]
@siauderman nope, no hack yet.

**Comment by KurchenkoDarij** [comment]
your snap works on my PC, how did you fix this problem?

**Comment by siauderman** [comment]
> your snap works on my PC, how did you fix this problem?

Lol, I didn't even do anything!

Will try again soon then, since it's working for you.

**Comment by siauderman** [comment]
> your snap works on my PC, how did you fix this problem?

I just tried reinstalling and it still didn't work. How did you manage to get it working?

**Comment by siauderman** [comment]
Hi it still doesn't work. Ubuntu impish 21.10. Upgraded to electron-builder 22.14.5. Again, running <app-name> from the command line does nothing with no errors. Same with clicking the icon. Running /snap/<app-name>/current/<app-name> works.

**Comment by siauderman** [comment]
I have already done a PR, yet the fix is not implemented.

This is very easy to fix, please just accept the PR.

**Comment by CrossyAtom46** [comment]
2024 and still same problem
